alpa/device_mesh.py

```diff
@@ -525,30 +634,76 @@ def run_resharding_recv_task(self, uuid, buf_uuids, set_empty_buffer=True):
                                          recv_spec.shape, recv_spec.dtype)
             for recv_tile_spec in recv_spec.tile_specs:
                 recv_tile_spec: ReshardingTileSpec
-                self.recv_tile(buf_uuid, recv_spec.device_id,
-                               recv_tile_spec.offset, recv_tile_spec.rank,
-                               recv_tile_spec.gpu_idx, task.group_name)
+                self.recv_tile_func(buf_uuid, recv_spec.device_id,
+                                    recv_tile_spec.offset, recv_tile_spec.rank,
+                                    recv_tile_spec.gpu_idx, task.group_name)
 
     def put_resharding_allgather_task(self, uuid, tasks):
         all_gather_task = ReshardingAllGatherTask(tasks)
         allgather_specs = all_gather_task.allgather_specs
         for allgather_spec in allgather_specs:
             device_ids = sorted(allgather_spec.device_ids)
             if repr(device_ids) not in self.allgather_communicators:
-                communicators = nccl.NcclCommunicator.initAll(list(device_ids))
-                self.allgather_communicators[repr(device_ids)] = communicators
+                if global_config.nccl_mode == "xla_extension":
+                    self.allgather_communicators[repr(device_ids)] = (
```

Comment:
Update the definition of `nccl_local_allgather_init_comms` in L165 and remove this if/else check.
```python3
self.nccl_local_allgather_init_comms = lambda x: xe.nccl_init_communicator(x, ENV.NCCL_USE_MULTISTREAM.val))
```

Reply:
OK